lib/rspec/core/bisect/runner.rb

```diff
@@ -71,24 +72,45 @@ def run_locations(*capture_args)
         # https://github.com/jruby/jruby/issues/2766
         if Open3.respond_to?(:capture2e) && !RSpec::Support::Ruby.jruby?
           def run_command(cmd)
-            Open3.capture2e(cmd).first
+            Open3.capture2e(bisect_environment_hash, cmd).first
           end
         else # for 1.8.7
           # :nocov:
           def run_command(cmd)
             out = err = nil
 
+            original_spec_opts = ENV['SPEC_OPTS']
+            ENV['SPEC_OPTS'] = spec_opts_without_bisect
+
```

Comment:
👍 did you try using the open3 feature of [passing in the env opts](http://ruby-doc.org/stdlib-2.2.3/libdoc/open3/rdoc/Open3.html#method-c-popen3)?

> The parameters env, cmd, and opts are passed to [Process.spawn](http://ruby-doc.org/core-2.2.3/Process.html#method-c-spawn)

> > env: hash
  name => val : set the environment variable
  name => nil : unset the environment variable
> > options: hash
  clearing environment variables:
    :unsetenv_others => true   : clear environment variables except specified by env
    :unsetenv_others => false  : don't clear (default)
> > If a hash is given as env, the environment is updated by env before exec(2) in the child process. If a pair in env has nil as the value, the variable is deleted.

> > set FOO as BAR and unset BAZ.
pid = spawn({"FOO"=>"BAR", "BAZ"=>nil}, command)

> > The :unsetenv_others key in options specifies to clear environment variables, other than specified by env.

> > pid = spawn(command, :unsetenv_others=>true) # no environment variable
pid = spawn({"FOO"=>"BAR"}, command, :unsetenv_others=>true) # FOO only